### PR TITLE
Fix resource leaks

### PR DESCRIPTION
**Pull request by JaimeBea**

### CODE REVIEW

**Comment by FrancPS** [comment]
Testing on Level1.scene, the moment Fang takes out her guns, this assert triggers.

Sin título

**Comment by JaimeBea** [comment]
> Testing on Level1.scene, the moment Fang takes out her guns, this assert triggers.
> 
> Sin título

This is the rotation bug. It's fixed in the rotation PR.

**Comment by FrancPS** [comment]
> This is the rotation bug. It's fixed in the rotation PR.

Then, is it ok if we merge this PR after the rotation fix, so we can correctly test if everything works fine?

**Comment by magalenyo** [comment]
> > This is the rotation bug. It's fixed in the rotation PR.
> 
> Then, is it ok if we merge this PR after the rotation fix (and merge it here), so we can correctly test if everything works fine?

It is a known "bug", testing can and SHOULD be done anyway. Just comment the line so it doesn't stop and that's it.

**Review by magalenyo — APPROVED** [review]
Reviewed and tested only the gameplay. No bugs apparently on Play, I'd appreciate an exhaustive testing on Game and on Engine: adding new resources, assigning, etc.

**Review by lowysole — APPROVED** [review]

**Review by lowysole — APPROVED** [review]

**Review by devildrake — APPROVED** [review]

**Review by magalenyo — DISMISSED** [review]

File: Project/Source/Resources/ResourceMesh.cpp [hunk 1]
```diff
@@ -12,6 +12,22 @@
...
+
+const unsigned positionSize = sizeof(float) * positionsSizeQuantity;
+const unsigned normalSize = sizeof(float) * normalSizeQuantity;
+const unsigned tangentSize = sizeof(float) * tangentSizeQuantity;
+const unsigned uvSize = sizeof(float) * uvSizeQuantity;
+const unsigned bonesIDSize = sizeof(unsigned) * bonesIDSizeQuantity;
+const unsigned weightsSize = sizeof(float) * weightSizeQuantity;
+
+const unsigned nonPosVertexAttributesSize = normalSize + tangentSize + uvSize + bonesIDSize + weightsSize;
+
```
Comment (Jordi-Gil) [inline Project/Source/Resources/ResourceMesh.cpp:30]: All these variables could be `constexpr` and be allocated at compile time.

**Review by Jordi-Gil — APPROVED** [review]

**Pull request merged**